dataplicity/client.py

```diff
@@ -153,74 +152,79 @@ def _sync(self):
 
         finally:
             elapsed = time.time() - start
-            log.debug('sync complete %0.2fs', elapsed)
+            log.debug("sync complete %0.2fs", elapsed)
 
     def _sync_meta(self, batch):
         """Sync meta information regarding host device."""
         try:
             meta = device_meta.get_meta()
             log.debug("syncing meta %r", meta)
         except:
-            log.exception('error getting meta')
+            log.exception("error getting meta")
         else:
             batch.call_with_id(
-                'set_agent_version_result',
-                'device.set_agent_version',
-                agent_version=meta['agent_version']
+                "set_agent_version_result",
+                "device.set_agent_version",
+                agent_version=meta["agent_version"],
             )
             batch.call_with_id(
-                'set_machine_revision_result',
-                'device.set_machine_revision',
-                revision_code=meta['machine_revision']
+                "set_machine_revision_result",
+                "device.set_machine_revision",
+                revision_code=meta["machine_revision"],
             )
             batch.call_with_id(
-                'set_os_version_result',
-                'device.set_os_version',
-                os_version=meta['os_version']
+                "set_os_version_result",
+                "device.set_os_version",
+                os_version=meta["os_version"],
             )
             batch.call_with_id(
-                'set_uname_result',
-                'device.set_uname',
-                uname=meta['uname']
+                "set_uname_result", "device.set_uname", uname=meta["uname"]
             )
             batch.call_with_id(
-                'set_ip_addresses',
-                'device.set_ip_addresses',
-                ip_list=meta['ip_list']
+                "set_ip_addresses_result",
+                "device.set_ip_addresses",
+                ip_list=meta["ip_list"],
+            )
+            batch.call_with_id(
+                "set_machine_defined_tags_result",
+                "device.set_machine_defined_tags",
+                tag_list=meta["tag_list"],
             )
 
     def _check_meta(self, batch):
         """Check previously sent meta information."""
-        log.debug('checking meta')
+        log.debug("checking meta")
         if self._sent_meta:
-            log.debug('meta was previously sent')
+            log.debug("meta was previously sent")
             return
         try:
             batch.check(
-                'set_agent_version_result',
-                'set_machine_revision_result',
-                'set_os_version_result',
-                'set_uname_result'
+                "set_agent_version_result",
+                "set_machine_revision_result",
+                "set_os_version_result",
+                "set_uname_result",
+                "set_ip_addresses_result",
```

Comment:
Was this not being checked then? So its feasible that an IP address may be missed if something goes wrong server side.

Reply:
I'll be honest, I don't know what that actually does.  I just noticed that the result wasn't there and probably should be.

Reply:
It throws an exception if any of the jsonrpc calls in the batch returned an error. So you probably do want them all in the list.